html/modules/custom/reliefweb_analytics/src/Command/ReliefwebMostReadCommand.php

```diff
@@ -0,0 +1,468 @@
+  }
+
+  /**
+   * Common payload.
+   */
+  protected function buildPayload($filter = NULL) {
+    $expressions = [
+      new FilterExpression([
+        'filter' => new Filter([
+          'field_name' => 'customEvent:content_type',
+          'string_filter' => new StringFilter([
+            'value' => 'Report',
+          ]),
+        ]),
+      ]),
+      new FilterExpression([
+        'filter' => new Filter([
+          'field_name' => 'pagePath',
+          'string_filter' => new StringFilter([
+            'match_type' => MatchType::BEGINS_WITH,
+            'value' => '/report',
+          ]),
+        ]),
+      ]),
+    ];
+
+    if ($filter) {
+      $expressions[] = $filter;
+    }
+
+    return [
+      'property' => 'properties/291027553',
+      'limit' => 5,
+      'returnPropertyQuota' => TRUE,
+      'dateRanges' => [
+        new DateRange([
+          'start_date' => '30daysAgo',
+          'end_date' => 'today',
+        ]),
+      ],
+      'dimensions' => [
+        new Dimension(['name' => 'pagePath']),
+      ],
+      'metrics' => [
+        new Metric(['name' => 'activeUsers']),
+      ],
+      'order_bys' => [
+        new MetricOrderBy(['metric_name' => 'screenPageViews']),
+      ],
+      'dimensionFilter' => new FilterExpression([
+        'and_group' => new FilterExpressionList([
+          'expressions' => $expressions,
+        ]),
+      ]),
+    ];
+  }
+
+  /**
+   * Homepage payload.
+   */
+  protected function getHomePagePayload() {
+    return $this->buildPayload();
+  }
+
+  /**
+   * Country payload.
+   */
+  protected function getCountryPayload($country_name) {
+    $filter = new FilterExpression([
+      'filter' => new Filter([
+        'field_name' => 'customEvent:content_report_primary_country',
+        'string_filter' => new StringFilter([
+          'value' => $country_name,
+        ]),
+      ]),
+    ]);
+    $payload = $this->buildPayload($filter);
+
+    return $payload;
+  }
+
+  /**
+   * Disaster payload.
+   */
+  protected function getDisasterPayload($disaster_name) {
+    $filter = new FilterExpression([
+      'filter' => new Filter([
+        'field_name' => 'customEvent:content_report_disaster',
+        'string_filter' => new StringFilter([
+          'value' => $disaster_name,
+        ]),
+      ]),
+    ]);
+    $payload = $this->buildPayload($filter);
+
+    return $payload;
+  }
+
+  /**
+   * Get all disasters payload.
+   */
+  protected function getAllDisastersPayload() {
+    $filter = new FilterExpression([
+      'filter' => new Filter([
+        'field_name' => 'customEvent:content_report_disaster',
+        'string_filter' => new StringFilter([
+          'value' => '.',
+          'match_type' => MatchType::PARTIAL_REGEXP,
+        ]),
+      ]),
+    ]);
+    $payload = $this->buildPayload($filter);
+
+    // Add dimension for disaster.
+    $payload['dimensions'][] = new Dimension(['name' => 'customEvent:content_report_disaster']);
```

Comment:
I'm guessing this is to be replaced with `dimension24`.

Reply:
Yes, left it for testing